src/utils.jl

```diff
@@ -13,7 +13,7 @@ function apply(f, args...; kwargs...)
     return f(args...; kwargs...)
 end
 
-function make_mlir_fn(mod, f, args, kwargs, name="main", concretein=true)
+function make_mlir_fn(mod, f, args, kwargs, name="main", concretein=true; toscalar=false)
```

Comment:
Maybe we could turn `name` and `concretein` into kwargs?
```suggestion
function make_mlir_fn(mod, f, args, kwargs; name="main", concretein=true, toscalar=false)
```